js/scripts/three-class-config.js

```diff
@@ -74,6 +74,14 @@ module.exports = {
     PropertyMixer: {
         relativePath: './animation/PropertyMixer',
     },
+    GLTFAsset: {
+        relativePath: './textures/GLTFAsset',
+        properties: {
+            scene: new Types.ThreeType('Scene', {use_instancedict: true}),
```

Comment:
Will this always produce a scene? If so, maybe this should inherit `Scene` instead. It will simplify the logic a lot,

Reply:
In my use case it will, same with many others use cases I suspect, so could simplify it to inherit from that, and just name it GLTFScene which would simplify it to just support that

Reply:
Yeah, that would be my recommendation.

Reply:
If you need to interact/modify the loaded model in Python, you would probably need to implement the loader in Python. Otherwise you are likely to end up in an asynchronous mess where the Python code has to wait for the JS to do stuff and sync it back before being able to continue executing.

Reply:
Unfortunately I need to be able interact with the scene that is loaded, so I'm fine with having python wait for a model loading, after doing some more work I have identified a few more issues

* `createUninitializedChildren` only works on `InitializedThreeType` parameters, for something being created in JS first this should ideally be all properties
* `createUninitializedChildren` does not handle nested properties (and handling this also means handling the allow_single arrays)
* (need to also verify impact) `utils.createModel` does not update the `initPromise` to include its additional setup for comms
* Not all properties exist to be pushed from JS to python (e.g. `_ref_geometry`), so `syncToModel` fails because it can't use `obj[propName]` on these
* (need to verify) Some nested traitlets appear to fail because the widget serialization is on the container not the instance

I've managed to use the mesh and parts that get loaded and pushed from this, however I've hit some very unusual issues where the data does not exist in python until I refresh the page. 

Unfortunately I need to stop working on this for the time being, so anyone feel free to take over, I may return to it in a few weeks.